CHANGELOG.md

```diff
@@ -1,3 +1,45 @@
+## 10.0.0 (YYYY-MM-DD)
+
+NOTE: This is a unified release note covering all v10.0.0-BETA.X v10.0.0-RC.X releases.
+
+NOTE: Support for syncing with realm.cloud.io and/or Realm Object Server has been replaced with support for syncing with MongoDB Realm Cloud.
+
+NOTE: This version uses the Realm file format to version 20. It is not possible to downgrade to earlier versions than v10.0.0-BETA.7. Non-sync Realms will be upgraded automatically. Synced Realms can only be automatically upgraded if created with Realm JavaScript v10.0.0-BETA.1 and above.
+
+### Breaking Changes
+* [RealmApp] Most API's for interacting with Realm Cloud has changed significantly. All new API's can be found in the `io.realm.mongodb` package. The entry point is though the `App` class from which you can create and login users and otherwise interact with MongoDB Realm. See [the docs](https://docs.mongodb.com/realm/android/) for further details. Synced Realms still use a `SyncConfiguration` that are largely created the same way.
+* [RealmApp] Client Resets are now handled through a custom `SyncConfiguration.Builder.clientResetHandler()` instead of through the default session error handler `SyncConfiguration.Builder.errorHandler()`
+* [RealmApp] Realm files have changed location on disk. They are now located in `getFiles()/mongodb-realm`.
```

Comment:
Missing requirement for a PK called `_id` on all top-level objects when using Sync. While the entirety of Sync is a breaking change, that one's worth calling out since it may not be obvious why a schema previously worked but it doesn't anymore.